docs/Deprecating-Disabling-and-Removing-Casks.md

```diff
@@ -0,0 +1,99 @@
+# Deprecating, Disabling and Removing Casks
+
+There are many reasons why casks may be deprecated, disabled or removed. This document explains the differences between each method as well as explaining when one method should be used over another.
+
+## Overview
+
+These general rules of thumb can be followed:
+
+- `deprecate!` should be used for casks that _should_ no longer be used.
+- `disable!` should be used for casks that _cannot_ be used.
+- Casks that are no longer acceptable in `homebrew/cask` or have been disabled for over a year _should_ be removed.
+
+## Deprecation
+
+If a user attempts to install a deprecated cask, they will be shown a warning message but the install will proceed.
+
+A cask should be deprecated to indicate to users that the cask should not be used and will be disabled in the future. Deprecated casks should continue to be maintained by the Homebrew maintainers if they continue to be installable. If this is not possible, they should be immediately disabled.
+
+The most common reasons for deprecation are when the upstream project is deprecated, unmaintained or archived.
+
+Casks should only be be deprecated if at least one of the following are true:
+
+- the cask does not install on any of our supported macOS versions
+- the cask has outstanding CVEs
+- the cask has [zero installs in the last 90 days](https://formulae.brew.sh/analytics/cask-install/90d/)
```

Comment:
We currently have a reason of `:discontinued` (upstream EOL) which matches none of these conditions.

Reply:
@Bo98 then we can adjust the messaging we use until we add more conditions. Can you suggest a wording change if you consider this feedback blocking?

Reply:
```suggestion
Casks should only be be deprecated if at least one of the following are true:

- the cask does not install on any of our supported macOS versions
- the cask has outstanding CVEs
- the cask has [zero installs in the last 90 days](https://formulae.brew.sh/analytics/cask-install/90d/)
- the software installed has been discontinued upstream
```

Reply:
yeah, adding an entry specifying deprecation by the upstream makes sense to me.

----

I wonder if we should also converge the usage into `:unmaintained` (more straightforward and same as `core`)